packages/astro/src/core/create-vite.ts

```diff
@@ -170,7 +172,31 @@ export async function createVite(
 	//   3. integration-provided vite config, via the `config:setup` hook
 	//   4. command vite config, passed as the argument to this function
 	let result = commonConfig;
-	result = vite.mergeConfig(result, settings.config.vite || {});
+	// command will be undefined when running sync
+	if (command) {
+		let plugins = settings.config.vite?.plugins;
+		if (plugins) {
+			const { plugins: _, ...rest } = settings.config.vite
+			const applyToFilter = command === 'build' ? 'serve' : 'build'
+			const applyArgs = [{...settings.config.vite, mode}, { command, mode }]
+			// @ts-ignore we know what are doing
```

Comment:
Please use `@ts-expect-error`, so we can remove the comment if the compiler gets better.

Would mind changing the comment too? Just saying "we know what we are doing" doesn't seem useful to other developers. Can we explain _what_ we are doing?

Reply:
![imagen](https://user-images.githubusercontent.com/6311119/221611283-6ba16668-5c42-41b5-8889-e2e5464e468f.png)

Reply:
we need to traverse flat plugins array to filter them